crates/cli/src/commands/build.rs

```diff
@@ -21,225 +22,432 @@ use openvm_transpiler::{elf::Elf, openvm_platform::memory::MEM_SIZE};
 use crate::{
     default::{
         DEFAULT_APP_CONFIG_PATH, DEFAULT_APP_EXE_PATH, DEFAULT_COMMITTED_APP_EXE_PATH,
-        DEFAULT_EXE_COMMIT_PATH, DEFAULT_MANIFEST_DIR,
+        DEFAULT_EXE_COMMIT_PATH,
     },
-    util::read_config_toml_or_default,
+    util::{find_manifest_dir, read_config_toml_or_default},
 };
 
 #[derive(Parser)]
 #[command(name = "build", about = "Compile an OpenVM program")]
 pub struct BuildCmd {
     #[clap(flatten)]
     build_args: BuildArgs,
+
+    #[clap(flatten)]
+    cargo_args: BuildCargoArgs,
 }
 
 impl BuildCmd {
     pub fn run(&self) -> Result<()> {
-        build(&self.build_args)?;
+        build(&self.build_args, &self.cargo_args)?;
         Ok(())
     }
 }
 
 #[derive(Clone, Parser)]
 pub struct BuildArgs {
-    #[arg(
-        long,
-        help = "Path to the directory containing the Cargo.toml file for the guest code (relative to the current directory)",
-        default_value = DEFAULT_MANIFEST_DIR
-    )]
-    pub manifest_dir: PathBuf,
-
-    #[arg(long, help = "Path to the target directory")]
-    pub target_dir: Option<PathBuf>,
-
-    #[arg(long, value_delimiter = ',', help = "Feature flags passed to cargo")]
-    pub features: Vec<String>,
-
-    #[clap(flatten, help = "Filter the target to build")]
-    pub bin_type_filter: BinTypeFilter,
-
     #[arg(
         long,
         default_value = "false",
-        help = "Skips transpilation into exe when set"
+        help = "Skips transpilation into exe when set",
+        help_heading = "OpenVM Options"
     )]
     pub no_transpile: bool,
 
     #[arg(
         long,
         default_value = DEFAULT_APP_CONFIG_PATH,
-        help = "Path to the SDK config .toml file that specifies the transpiler extensions"
+        help = "Path to the SDK config .toml file that specifies the transpiler extensions",
```

Comment:
```suggestion
        help = "Path to the OpenVM config .toml file that specifies the VM extensions",
```
I think we use it more generally than just transpiler